spec/models/grade_entry_form_spec.rb

```diff
@@ -0,0 +1,241 @@
+require 'spec_helper'
+require File.expand_path(File.join(File.dirname(__FILE__), '..', '..', 'test', 'test_helper'))
+require File.expand_path(File.join(File.dirname(__FILE__), '..', '..', 'test', 'blueprints', 'blueprints'))
+
+describe GradeEntryForm do
+
+  # Basic validation tests
+  it { is_expected.to have_many(:grade_entry_items) }
+  it { is_expected.to have_many(:grade_entry_students) }
+  it { is_expected.to have_many(:grades) }
+  it { is_expected.to validate_presence_of(:short_identifier) }
+
+  # Dates in the past should also be allowed
+  it { is_expected.to allow_value(1.day.ago).for(:date) }
+  it { is_expected.to allow_value(1.day.from_now).for(:date) }
+  it { is_expected.not_to allow_value('100-10').for(:date) }
+  it { is_expected.not_to allow_value('abcd').for(:date) }
+
+  context ' A good Grade entry model' do
```

Comment:
This `context` block is unnecessary. The test can be rewritten in the same style as the ones above.